tests/net/lib/coap/src/main.c

```diff
@@ -66,6 +74,47 @@ static struct sockaddr_in6 dummy_addr = {
 	.sin6_family = AF_INET6,
 	.sin6_addr = peer_addr };
 
+
+
+static void get_from_ip_addr(struct coap_packet *cpkt,
+			     struct sockaddr_in6 *from)
+{
+	struct net_udp_hdr hdr, *udp_hdr;
+
+	udp_hdr = net_udp_get_hdr(cpkt->pkt, &hdr);
+	if (!udp_hdr) {
+		return;
+	}
+
+	net_ipaddr_copy(&from->sin6_addr, &NET_IPV6_HDR(cpkt->pkt)->src);
+	from->sin6_port = udp_hdr->src_port;
+	from->sin6_family = AF_INET6;
+}
+
+static int well_known_core_get(struct coap_resource *resource,
+			       struct coap_packet *request)
+{
+	struct coap_packet response;
+	struct sockaddr_in6 from;
+	struct net_pkt *pkt;
+	struct net_buf *frag;
+	int r;
+
+	static struct net_context *context;
+
+	pkt = net_pkt_get_tx(context, K_FOREVER);
+	frag = net_pkt_get_data(context, K_FOREVER);
+	net_pkt_frag_add(pkt, frag);
+
+	r = coap_well_known_core_get(resource, request, &response, pkt);
```

Comment:
Do not only concentrate on code coverage, these are unit tests. Also test some parsed information like parsed and expected information is matching or not.